.circleci/config.yml

```diff
@@ -391,7 +391,44 @@ jobs:
       - run: source ${BASH_ENV} && make deps
       - run: brew install protobuf
       - run: PATH="$GOPATH/bin:${HOME}/goinstall/go/bin:$PATH" make generate-structs
-      - run: source ${BASH_ENV} && make pkg/darwin_amd64.zip
+
+      - run:
+          name: prepare ui cache
+          command: git log -n 1 --pretty=format:%H ui > /tmp/ui-sha
+
+      - restore_cache:
+          name: restore compiled ui assets
+          keys:
+          - v1-binary-ui-assets-darwin-{{ checksum "/tmp/ui-sha" }}
```

Comment:
FYI - caching is necessary here.  Without a cache, installing npm dependencies and compiling the ui assets adds ~7 minutes in macOS: https://app.circleci.com/pipelines/github/hashicorp/nomad/14254/workflows/84c1a4b5-9d99-4e8e-b384-6fceee403dd3/jobs/131485